front/src/modules/timesStops/hooks/useOutputTableData.tsx

```diff
@@ -0,0 +1,111 @@
+import { useMemo } from 'react';
+
+import { keyBy } from 'lodash';
+import { useSelector } from 'react-redux';
+
+import type {
+  PathPropertiesFormatted,
+  SimulationResponseSuccess,
+} from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { formatSuggestedOperationalPoints } from 'modules/pathfinding/utils';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import { getSelectedTrainId } from 'reducers/osrdsimulation/selectors';
+import { ISO8601Duration2sec, datetime2sec, secToHoursString } from 'utils/timeManipulation';
+
+import useComputedMargins from './useComputedMargins';
+import { findNextScheduledPoint } from '../utils';
+
+function useOutputTableData(
+  simulatedTrain: SimulationResponseSuccess,
+  pathProperties: PathPropertiesFormatted,
+  operationalPoints: OperationalPointWithTimeAndSpeed[],
+  startTime: string
+) {
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  const { data: selectedTrainSchedule = {} as TrainScheduleResult } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
+      {
+        id: selectedTrainId as number,
+      },
+      { skip: !selectedTrainId }
+    );
+
+  const trainScheduleByAt = keyBy(selectedTrainSchedule.schedule, 'at');
```

Comment:
I thought at first that schedulesByAt was containing trainSchedules and not only schedules
```suggestion
  const schedulesByAt = keyBy(selectedTrainSchedule.schedule, 'at');
```

Reply:
done